quiz-app-api/app/routes/middlewares/auth.js

```diff
@@ -0,0 +1,30 @@
+const jwt = require('jsonwebtoken');
+const config = require('../../../config/config')
+const { AUTH_HEADER } = require('../../utils/constants')
+
+const authenticateUser = (req, res, next) => {
+  console.log(req.headers);
+  console.log(req.headers[AUTH_HEADER])
+  console.log(req.headers[AUTH_HEADER].split(' ')[1]);
+  if (req.headers && req.headers[AUTH_HEADER]) {
+    const token = req.headers[AUTH_HEADER].split(' ')[1]
+    jwt.verify(token, config.jwt_secret, (error, decoded) => {
+      if (error) {
+        res.status(401).json({
+          error,
+          message: 'Invalid Token! Please sign in to access this route'
+        })
+      }
+
+      // TODO: verfiy if the decoded contains a valid userId and only then call next()
```

Comment:
Should we verify the user from the database here? This will happen for every incoming request for protected routes.